consensus/broadcast/shardChainMessenger_test.go

```diff
@@ -512,3 +566,13 @@ func TestShardChainMessenger_BroadcastBlockDataLeaderShouldTriggerWaitingDelayed
 		assert.True(t, broadcastUsingPrivateKeyWasCalled.IsSet())
 	})
 }
+
+func TestShardChainMessenger_PrepareBroadcastHeaderValidatorShouldFailHeaderNil(t *testing.T) {
+
+	pkBytes := make([]byte, 32)
+	args := createDefaultShardChainArgs()
+
+	scm, _ := broadcast.NewShardChainMessenger(args)
+
+	scm.PrepareBroadcastHeaderValidator(nil, nil, nil, 1, pkBytes)
```

Comment:
at that time, the delayedBroadcaster was not mocked. Now is updated

Reply:
also, altough it is just a test, we want to avoid panics. This code may panic if somethings was wrong with the args. It will return a nil scm, which tnen will be called on L625. My suggestion is on L624 either add a require.NotNil(t, scm), either replace _ with err and check require.NoErr(t, err). Valid on almost all tests